src/management/ManagementTokenProvider.js

```diff
@@ -97,38 +131,6 @@ ManagementTokenProvider.prototype.getAccessToken = function() {
   }
 };
 
-ManagementTokenProvider.prototype.getCachedAccessToken = Promise.promisify(
-  memoizer({
-    load: function(options, callback) {
-      this.clientCredentialsGrant(options.domain, options.scope, options.audience)
```

Comment:
I have no idea how this worked before, but it wasn't supposed to (`this.clientCredentialsGrant` is undefined because `this` is not bound to the class, it's bound to the memoizer object). This was the reason the tests failed in newer versions of nodejs, so I fixed this as well.